### PR TITLE
customize: add names vocabulary docs

### LINKED ISSUE #288: names documentation

### PR DESCRIPTION
**Pull request by ppanero**
- closes #288

### CODE REVIEW

File: docs/customize/vocabularies/names.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+## Creating your names.yaml file
+
+The Names vocabulary has been implemented with the 
+[ORCiD dataset](https://orcid.figshare.com/articles/dataset/ORCID_Public_Data_File_2021/16750535?file=31020067)
+in mind. This means that there is built-in functionality to process it already
+implemented. For example, you can use the **convert** command to change this
+dataset into a yaml file with the appropriate format.
+
+```shell
+invenio vocabularies convert -v names -o "/Users/ppanero/Downloads/ORCID_2021_10_summaries.tar.gz" -t openaire-names.yaml
```
Comment (slint) [inline docs/customize/vocabularies/names.md:86]:
```suggestion
invenio vocabularies convert -v names -o "~/Downloads/ORCID_2021_10_summaries.tar.gz" -t names.yaml
```

To protect some innocent.

Comment (ppanero) [reply docs/customize/vocabularies/names.md:86]: oops 👼

File: docs/customize/vocabularies/names.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+There are several ways to import names records. The most straight forward
+is by definition file, similar to the old `vocabularies.yaml`.
+In this file you will define how the data source will be read, if it needs
+any processing and finally how they should be written. For the purpose of
+importing vocabularies it will most likely default to reading a yaml file
+and using a service API. An example of this new definition file, lets call
+it `vocabularies-future.yaml`.
+
+```yaml
+names:
```
[the remark this answers is not in the document]
Comment (ppanero) [reply docs/customize/vocabularies/names.md:55]: well... it does, with a different name, the default config wont be found [here](https://github.com/inveniosoftware/invenio-vocabularies/blob/master/invenio_vocabularies/cli.py#L25). It is not ideal... but I did not find a better option to do so atm (might come back again to the _using entry points_ topic....

File: docs/customize/vocabularies/names.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  affiliations:
+  - id: cern
+```
+
+## How to import and update your name records
+
+The Names vocabulary uses the new APIs for processing vocabularies:
+DataStreams. This means that name records will not be imported by the
+usual `invenio rdm-records fixtures` command. You can find more
+information about this new API in the [RFC](TODO).
```
[the remark this answers is not in the document]
Comment (slint) [reply docs/customize/vocabularies/names.md]: Added links to the RFC PRs

**Pull request merged**